Retain IronCurtain effect upon DeploysInto/UndeploysInto (#698)
Phobos-developers/Phobos: docs/Fixed-or-Improved-Logics.md

- Setting VehicleType `Speed` to 0 now makes game treat them as stationary, behaving in very similar manner to deployed vehicles with `IsSimpleDeployer` set to true. Should not be used on buildable vehicles, as they won't be able to exit factories.

> ### IronCurtain status inheritance on DeploysInto/UndeploysInto

Reviewer: ```suggestion
### Preserve Iron Curtain status on type conversion 
```

> - When an iron-curtained vehicle is deployed into a building and vice versa, the iron-curtain effect will be removed. This behavior now can be customized. The iron-curtain effect can be preserved.

Reviewer: ```suggestion
- Iron Curtain status is now preserved when converting between TechnoTypes.
  - This behavior can be customized on per-TechnoType and global basis.
```

> ![image](_static/images/preserve-ic.gif)

Reviewer: Should move it before the description.

- Setting VehicleType `Speed` to 0 now makes game treat them as stationary, behaving in very similar manner to deployed vehicles with `IsSimpleDeployer` set to true. Should not be used on buildable vehicles, as they won't be able to exit factories.

### Preserve Iron Curtain status on type conversion

![image](_static/images/preserve-ic.gif)

- Iron Curtain status is now preserved by default when converting between TechnoTypes via `DeploysInto`/`UndeploysInto`.
  - This behavior can be turned off per-TechnoType and global basis.
  - `IronCurtain.Modifier` is re-applied upon type conversion.

In `rulesmd.ini`:
```ini
[CombatDamage]

> IronCurtain.KeptOnDeploy=no  ; boolean

Reviewer: ```suggestion
IronCurtain.KeptOnDeploy=yes ; boolean
```
